lib/service/user_service.dart

```diff
@@ -0,0 +1,22 @@
+import 'dart:io';
+
+import 'package:backtrip/model/trip.dart';
+import 'package:backtrip/service/trip_service.dart';
+import 'package:backtrip/util/backtrip_api.dart';
+import 'package:flutter/foundation.dart';
+import 'package:http/http.dart' as http;
+
+class UserService {
+
+  static Future<List<Trip>> getTrips(userId) async {
+    var uri = '${BacktripApi.path}/user/${userId}/trips';
+    var headers = { HttpHeaders.authorizationHeader: BacktripApi.token };
+    final response = await http.get(uri, headers: headers);
+
+    if (response.statusCode == HttpStatus.ok) {
+      return compute(TripService.parseTrips, response.body);
```

Comment:
La fonction built-in compute() permet de lancer des fonctions avec des traitements "lourds" en arrière plan
https://flutter.dev/docs/cookbook/networking/background-parsing#4-move-this-work-to-a-separate-isolate